plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/HdfsInputFile.java

```diff
@@ -57,7 +57,8 @@ public long getLength()
     @Override
     public SeekableInputStream newStream()
     {
-        return environment.doAs(user, delegate::newStream);
+        // Hack: this wrapping is required to circumvent https://github.com/trinodb/trino/issues/5201
+        return new HdfsInputStream(environment.doAs(user, delegate::newStream));
```

Comment:
How does wrapping help alleviate #5201? what's the mechanics?

Reply:
@findepi does this help with mechanics? https://github.com/trinodb/trino/issues/5201#issuecomment-876583249

Reply:
i.e. the "rewrapping" is not done. https://github.com/apache/iceberg/blob/master/parquet/src/main/java/org/apache/iceberg/parquet/ParquetIO.java#L93